src/SourceCode.Clay.Data/SqlClient/SqlConnectionRetryOptions.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Diagnostics.Contracts;
+
+namespace SourceCode.Clay.Data.SqlClient
+{
+    /// <summary>
+    /// A property bag to hold various connection retry options.
+    /// </summary>
+    public sealed class SqlConnectionRetryOptions
```

Comment:
Struct. Use nullable backing fields and `.GetValueOrDefault` to provide default value. e.g.

```private byte? _connectRetryCount;
public byte ConnectRetryCount {
 get => _connectRetryCount.GetValueOrDefault(10);
 set => _connectRetryCount = value;
}